python_modules/dagster-test/dagster_test/fixtures/docker_compose.py

```diff
@@ -10,48 +10,42 @@
 from .utils import BUILDKITE
 
 
-@pytest.fixture(scope="module")
-def docker_compose_cm(test_directory):
-    @contextmanager
-    def docker_compose(
-        docker_compose_yml=None,
-        network_name=None,
-        docker_context=None,
-        service=None,
-    ):
-        if not docker_compose_yml:
-            docker_compose_yml = default_docker_compose_yml(test_directory)
-        if not network_name:
-            network_name = network_name_from_yml(docker_compose_yml)
-        try:
-            docker_compose_up(docker_compose_yml, docker_context, service)
-            if BUILDKITE:
-                # When running in a container on Buildkite, we need to first connect our container
-                # and our network and then yield a dict of container name to the container's
-                # hostname.
-                with buildkite_hostnames_cm(network_name) as hostnames:
-                    yield hostnames
-            else:
-                # When running locally, we don't need to jump through any special networking hoops;
-                # just yield a dict of container name to "localhost".
-                yield dict((container, "localhost") for container in list_containers())
-        finally:
-            docker_compose_down(docker_compose_yml, docker_context, service)
-
-    return docker_compose
+@contextmanager
```

Comment:
I think we can get the best of both worlds by keeping the fixture (which handles things like finding the dockerfile by default automatically) and also extracting the context manager. That'll decrease the number of places you need to make fixes in response to this PR and it'll keep usage of the fixtures simpler. Probably a good pattern for us to follow in general - thin fixtures that call separate context managers.